src/backend/distributed/connection/connection_configuration.c

```diff
@@ -281,8 +281,10 @@ GetConnParams(ConnectionHashKey *key, char ***keywords, char ***values,
 	for (paramIndex = 0; paramIndex < ConnParams.size; paramIndex++)
 	{
 		/* copy the keyword&value pointers to the new array */
```

Comment:
stale comment, we're now copying the strings since the originals might be free()-d when settings are reloaded.